docs/release/release-tasks.md

```diff
@@ -0,0 +1,384 @@
+# Release Tasks
+
```

Comment:
@CecileRobertMichon I left out the following

> Ensure only eligible PRs are cherry-picked to the active release branches

I think this should be the responsibility of the approvers merging the PRs against main. If there is something that should be communicated to the release team we can add it to the initial release note doc or communicate with the release team members.

Reply:
That makes sense. I think the release team should be _aware_ of cherry-picks getting merged but not necessarily own the decision-making of which ones are accepted. We could probably just cc the release team on cherry-pick bot PRs?

Reply:
> just cc the release team on cherry-pick bot PRs

No objections, but the question is how. 

As of today the cherry-pick bot doesn't allow customizing the PR description doing it manually is too much effort and easily missed.

I would suggest tracking a follow-up issue to improve the cherry-picker (it's not super-trivial but probably doable)

Reply:
Tracked under improvement tasks in https://github.com/kubernetes-sigs/cluster-api/issues/7389

Reply:
We could probably also do this easily with a GitHub action. Will add a note